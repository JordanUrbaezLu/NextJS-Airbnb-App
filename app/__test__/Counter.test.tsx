import { render, screen } from "@testing-library/react";
import Counter from "../components/inputs/Counter";

describe("Counter", () => {
  test("Should render component correctly.", () => {
    const { container } = render(
      <Counter
        onChange={jest.fn()}
        title="Airbnb"
        value={0}
        subtitle="subtitle"
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  describe("Interaction", () => {
    test("Should trigger onChange on add and decrease", () => {
      const onChange = jest.fn();

      const { rerender } = render(
        <Counter
          onChange={onChange}
          title="Airbnb"
          value={0}
          subtitle="subtitle"
        />
      );
      const subtractButton = screen.getAllByRole("button")[0];
      const addButton = screen.getAllByRole("button")[1];

      subtractButton.click();
      expect(onChange).toHaveBeenCalledTimes(1);

      addButton.click();
      expect(onChange).toHaveBeenCalledTimes(2);

      rerender(
        <Counter
          onChange={onChange}
          title="Airbnb"
          value={1}
          subtitle="subtitle"
        />
      );

      subtractButton.click();
      expect(onChange).toHaveBeenCalledTimes(2);
    });
  });
});
