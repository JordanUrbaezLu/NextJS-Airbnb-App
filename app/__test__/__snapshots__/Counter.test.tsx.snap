// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Should render component correctly. 1`] = `
<div
  class="flex flex-row items-center justify-between"
>
  <div
    class="flex flex-col"
  >
    <div
      class="font-medium"
    >
      Airbnb
    </div>
    <div
      class="font-light text-gray-600"
    >
      subtitle
    </div>
  </div>
  <div
    class="flex flex-row items-center gap-4"
  >
    <div
      class=" flex h-10 w-10 cursor-pointer items-center justify-center rounded-full border-[1px] border-neutral-400 text-neutral-600 transition hover:opacity-80 "
      role="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M872 474H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h720c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
        />
      </svg>
    </div>
    <div
      class=" text-xl  font-light  text-neutral-600 "
    >
      0
    </div>
    <div
      class=" flex h-10 w-10 cursor-pointer items-center justify-center rounded-full border-[1px] border-neutral-400 text-neutral-600 transition hover:opacity-80 "
      role="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        t="1551322312294"
        version="1.1"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M474 152m8 0l60 0q8 0 8 8l0 704q0 8-8 8l-60 0q-8 0-8-8l0-704q0-8 8-8Z"
        />
        <path
          d="M168 474m8 0l672 0q8 0 8 8l0 60q0 8-8 8l-672 0q-8 0-8-8l0-60q0-8 8-8Z"
        />
      </svg>
    </div>
  </div>
</div>
`;
